src/elements/content-sidebar/ActivitySidebarFilter.js

```diff
@@ -9,29 +9,57 @@ import { FormattedMessage } from 'react-intl';
 import DropdownMenu, { MenuToggle } from '../../components/dropdown-menu';
 import PlainButton from '../../components/plain-button';
 import messages from './messages';
-import { COMMENT_STATUS_OPEN } from '../../constants';
+import {
+    ACTIVITY_FILTER_OPTION_ALL,
+    ACTIVITY_FILTER_OPTION_OPEN,
+    ACTIVITY_FILTER_OPTION_RESOLVED,
+    ACTIVITY_FILTER_OPTION_TASKS,
+    COMMENT_STATUS_OPEN,
+    COMMENT_STATUS_RESOLVED,
+    FEED_ITEM_TYPE_TASK,
+} from '../../constants';
 import { Menu, SelectMenuItem } from '../../components/menu';
-import type { FeedItemStatus } from '../../common/types/feed';
+import type { ActivityFilterOption, ActivityFilterStatus } from '../../common/types/feed';
 import './ActivitySidebarFilter.scss';
 
 type ActivitySidebarFilterProps = {
-    feedItemStatus?: FeedItemStatus,
-    onFeedItemStatusClick: (status?: FeedItemStatus) => void,
+    activityFilterOptions: ActivityFilterOption[],
+    feedItemStatus?: ActivityFilterStatus,
+    onFeedItemStatusClick: (status?: ActivityFilterStatus) => void,
 };
 
-const ACTIVITY_FEED_ITEM_ALL = 'all';
-const ACTIVITY_FEED_ITEM_OPEN = COMMENT_STATUS_OPEN;
-
-const statuses = [ACTIVITY_FEED_ITEM_ALL, ACTIVITY_FEED_ITEM_OPEN];
-const statusMap = {
-    [ACTIVITY_FEED_ITEM_ALL]: { msg: messages.activitySidebarFilterOptionAll, val: undefined },
-    [ACTIVITY_FEED_ITEM_OPEN]: { msg: messages.activitySidebarFilterOptionOpen, val: COMMENT_STATUS_OPEN },
+const filterOptionToStatus = {
+    [ACTIVITY_FILTER_OPTION_ALL]: ACTIVITY_FILTER_OPTION_ALL,
+    [ACTIVITY_FILTER_OPTION_OPEN]: COMMENT_STATUS_OPEN,
+    [ACTIVITY_FILTER_OPTION_RESOLVED]: COMMENT_STATUS_RESOLVED,
+    [ACTIVITY_FILTER_OPTION_TASKS]: FEED_ITEM_TYPE_TASK,
 };
 
 function ActivitySidebarFilter({
-    feedItemStatus = ACTIVITY_FEED_ITEM_ALL,
+    activityFilterOptions,
+    feedItemStatus = ACTIVITY_FILTER_OPTION_ALL,
     onFeedItemStatusClick,
 }: ActivitySidebarFilterProps) {
+    const isCommentsOnlyFilter = !activityFilterOptions.includes(ACTIVITY_FILTER_OPTION_TASKS);
```

Comment:
We can have the ternary here return the all message, so we can keep logic out of the statusMap